### ISSUE TITLE
Name randomization of group.

### ISSUE THREAD

**Issue by AxiomOfChoices** [ElizabethHudnott/peerjs-groups#4]
So I've been trying to use this code for my project on the side and the main problem I'm trying to tackle is the admin disconnecting issue mentioned in #2. I've tried to solving this by having each client listen for the 'userleft' event and adding an extra parameter to that event that lest them know if it was the admin that left, after this each client disconnects from the group and attempts to reconnect, my thinking was that which ever client did this first would become the new group admin (as no other client would be connected) and then each other client would connect to it.

And while this does work it introduces a new problem where the new admin keeps its previous name(which is passed to the connect function) but when the other clients attempt to connect to it their names become randomizes regardless of what is passed to their connect function.

![image](https://user-images.githubusercontent.com/14252368/103449215-2de71100-4c73-11eb-828e-75bd56b84b8b.png)

I have been trying to check for a while what could be causing this and the only thing I narrowed down upon is that this happens if the clients try to connect within a small amount of time of each other(give or take 300ms).
What would be a possible cause of this issue?

**Comment by rohanbam** [comment]
@AxiomOfChoices, did you ever manage to resolve this issue?

**Comment by AxiomOfChoices** [comment]
@rohanbam Nope, but I also didn't understand much about the protocol at the time. It's possible the issue is quite simple and I just didn't see it.

**Comment by rohanbam** [comment]
Ah ok. Thanks @AxiomOfChoices. I'll see if I can find a way around it. Cheers.
